tests/spec_constants/spec_constants_common.h

```diff
@@ -48,11 +48,11 @@ constexpr int default_val = 20;
 
 template <typename T, int case_num>
 constexpr sycl::specialization_id<T> spec_const(
-    user_def_types::get_init_value_helper<T>(default_val));
+    user_def_types::init_value_helper<T>::get(default_val));
```

Comment:
What is the value of having `struct` instead of generic functions like
```suggestion
    user_def_types::get_init_value<T>(default_val));
```
?

Reply:
@keryell, I think @AlexeySachkov is seeking for discussion of this change in this comment: https://github.com/KhronosGroup/SYCL-CTS/pull/856#issuecomment-1895869936

> However, I do like the new name better, because with the old name one might think that function is returning some kind of helper instead of an init value. Let me know about the preference, I can rework the PR accordingly.

Reply:
I do not like the word `helper` which does not *help* the understanding. :-)

Reply:
> What is the value of having struct instead of generic functions like

Yes, that's a good point. We can still have it as a function, but we can also rename it along the way, I will update the PR

Reply:
Refactored back to function, removed `helper` from the name in 6211bfc04faed9ce593f601af3b1e925e6ea7d33